### PR TITLE
🐛 attempt to fix banded scoring with exceptions

### PR DESCRIPTION
**Pull request by imilchev**
Second attempt at fixing banded scoring, after #1350 didn't fully work

### CODE REVIEW

File: policy/score_calculator.go [hunk 1]
```diff
@@ -490,22 +468,22 @@ func (c *bandedScoreCalculator) Add(score *Score, impact *explorer.Impact) {
 
 			if category <= 10 {
 				c.critMax += score.Weight
-				if score.Value == 100 {
+				if score.Value < 100 {
```
Comment (imilchev) [inline policy/score_calculator.go:471]: based on my understanding the counter should only increase if the score is not `Pass`

File: policy/score_calculator_test.go [hunk 2]
```diff
@@ -239,7 +239,24 @@ func TestBandedScores(t *testing.T) {
...
+				{Value: 100, ScoreCompletion: 100, DataCompletion: 33, DataTotal: 3, Weight: 10, Type: ScoreType_Result},
+			},
+			impacts: []*explorer.Impact{
+				// 10 critical checks
+				{Value: &explorer.ImpactValue{Value: 100}},
+				{Value: &explorer.ImpactValue{Value: 100}},
+				// 10 high checks
+				{Value: &explorer.ImpactValue{Value: 80}},
+			},
+			out: &Score{Value: 100, ScoreCompletion: 100, DataCompletion: 66, Weight: 20, Type: ScoreType_Result},
```
Comment (imilchev) [inline policy/score_calculator_test.go:259]: I added a case with all passing checks. I would expect if everything passed, then the policy score should also be 100. In the original code that is not the case. It works now, however, the values for the other cases changed too...

File: policy/score_calculator.go [hunk 3]
```diff
@@ -67,28 +67,6 @@ func AddSpecdScore(calculator ScoreCalculator, s *Score, found bool, impact *exp
 		return
 	}
 
-	// everything else is modify or activate
-
-	if impact.Scoring == explorer.ScoringSystem_IGNORE_SCORE {
```
Comment (imilchev) [inline policy/score_calculator.go:72]: in #1350 I already made sure all the calculators handle exceptions properly, so this workaround shouldn't be needed anymore

File: policy/score_calculator.go (hunk 1 above)
Comment (jaym) [reply policy/score_calculator.go:471]: based on the `Calculate` function, I think `c.crit` and the rest are supposed to count the number of failures:
```
		if c.crit != 0 {
			res.Value = uint32(math.Floor(float64(50) * pcrCrit))
		} else if c.high != 0 {
			res.Value = uint32(math.Floor(float64(50)*pcrHigh)) + 10
```
If there failures, we need to fall into the first category with failures.

So this all seems right to me

**Review by jaym — APPROVED** [review]

**Pull request merged**